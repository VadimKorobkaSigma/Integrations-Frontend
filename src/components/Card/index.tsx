import * as React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

interface Props {
    children: JSX.Element | JSX.Element[];
    className?: string;
}

const Card: React.FC<Props> = ({ children, className }) => {
    return <div className={cn(styles.card, className)}>{children}</div>;
};

export default React.memo(Card);
