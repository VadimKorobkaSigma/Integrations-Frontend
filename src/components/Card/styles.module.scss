.card {
  display: grid;
  max-width: 1015px;
  padding: 0;
  box-shadow: rgba(0, 0, 0, 0.12) 0px 4px 10px;
  border-radius: 4px;
  overflow: hidden;
}
