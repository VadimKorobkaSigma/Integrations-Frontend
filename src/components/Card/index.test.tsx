import * as React from 'react';
import Card from './index';
import { shallow } from 'enzyme';

test('Card Component only with children', () => {
    const wrapper = shallow(
        <Card>
            <p id="child-node">Test</p>
        </Card>,
    );

    expect(wrapper.find('#child-node')).toBeDefined();
    expect(wrapper.html()).toMatchSnapshot();
});

test('Card Component with classname', () => {
    const wrapper = shallow(
        <Card className="test">
            <p id="child-node">Test</p>
        </Card>,
    );

    expect(wrapper.hasClass('test')).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
});
