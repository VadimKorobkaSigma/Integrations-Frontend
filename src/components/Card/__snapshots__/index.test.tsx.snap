// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component only with children 1`] = `"<div class=\\"\\"><p id=\\"child-node\\">Test</p></div>"`;

exports[`Card Component with classname 1`] = `"<div class=\\"test\\"><p id=\\"child-node\\">Test</p></div>"`;
